staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +197,29 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// IsStopped reports whether the informer has already been stopped.
+	// Adding event handlers to already stopped informers is not possible.
+	// An informer already stopped will never be started again.
+	IsStopped() bool
+}
+
+// ResourceEventHandlerHandle is a handle returned by the
+// registration methods of SharedInformers for a registered
+// ResourceEventHandler. It can be used later on to remove
+// a registration again.
+// This indirection is required, because the ResourceEventHandler
+// interface can be implemented by non-go-comparable handlers, which
+// could not be removed from a list anymore.
+type ResourceEventHandlerHandle struct {
```

Comment:
Can we call this a `ResourceEventRegistration`? "HandlerHandle" is really awkward to talk about.

Reply:
Ok, but it is nor a registration of a resource event. Then we should name it `ResourceEventHandlerRegistration`, because it is a registration of a (resource event) handler.